src/steps/to-hast.ts

```diff
@@ -19,12 +19,91 @@ import { Helix } from '@adobe/helix-universal';
 import { toHast as mdast2hast, defaultHandlers, State } from 'mdast-util-to-hast';
 import { raw } from 'hast-util-raw';
 import { mdast2hastGridTablesHandler, TYPE_TABLE } from '@adobe/mdast-util-gridtables';
+import type { Code } from 'mdast-util-to-hast/lib/handlers/code.js';
+import type { Element } from 'hast';
+
+export function code(state: State, node: Code): any {
+  const pre = defaultHandlers.code(state, node);
+  const meta = (node.meta ?? '').trim();
+
+  const codeAttrs: Record<string, string> = {};
+  const codeClasses: string[] = [];
+
+  // Robustly extract attributes and flags for <code>
+  const parseMeta = (input: string): Array<{ key: string; value?: string }> => {
+    const tokens: Array<{ key: string; value?: string }> = [];
+    let i = 0;
+    const n = input.length;
+    while (i < n) {
+      while (i < n && /\s/.test(input[i]!)) i++;
+      if (i >= n) break;
+      const keyStart = i;
+      while (i < n && !/\s|=/.test(input[i]!)) i++;
+      const key = input.slice(keyStart, i);
+      while (i < n && /\s/.test(input[i]!)) i++;
+      let value: string | undefined;
+      if (i < n && input[i] === '=') {
+        i++;
+        while (i < n && /\s/.test(input[i]!)) i++;
+        if (i < n && input[i] === '"') {
+          i++;
+          const valStart = i;
+          while (i < n && input[i] !== '"') i++;
+          value = input.slice(valStart, i);
+          if (i < n && input[i] === '"') i++;
+        } else {
+          const valStart = i;
+          while (i < n && !/\s/.test(input[i]!)) i++;
+          value = input.slice(valStart, i);
+        }
+      }
+      if (key) tokens.push({ key, value });
+    }
+    return tokens;
+  };
+
+  for (const { key, value } of parseMeta(meta)) {
+    if (key === 'disableLineNumbers' && value === undefined) {
```

Comment:
We need to handle `disableLineNumbers` and `disableLineNumbers={true}` the same way, since they are equivalent JSX syntax.

```suggestion
    if (key === 'disableLineNumbers' && (value === undefined || value === 'true')) {
```

For context, we have [similar syntax in Gatsby](https://github.com/adobe/aio-theme?tab=readme-ov-file#redoclyapiblock), e.g.

<img width="922" height="452" alt="Screenshot 2025-08-20 at 7 28 03 AM" src="https://github.com/user-attachments/assets/102257cc-5171-40cf-a86e-75e9a6197736" />